data_hub_api/main.py

```diff
@@ -29,6 +29,16 @@ def get_enhanced_preprints_docmaps_by_preprint_doi(preprint_doi: str):
             )
         return docmaps
 
+    @router.get("/v1/by_publisher/elife/get-by-doi")
+    def get_enhanced_preprints_docmaps_by_preprint_doi_by_publisher_elife(preprint_doi: str):
+        docmaps = docmaps_provider.get_docmaps_by_doi(preprint_doi)
+        if not docmaps:
+            raise HTTPException(
+                status_code=404,
+                detail="No Docmaps available for requested DOI from the publisher eLife"
+            )
+        return docmaps[0]
```

Comment:
very minor: would it make sense to put `assert len(docmaps) == 1` in front of it?